// @rarible/ethereum-sdk-test-common/build/create-e2e-provider.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createE2eProvider = void 0;
var form_data_1 = __importDefault(require("form-data"));
var web3_provider_engine_1 = __importDefault(require("web3-provider-engine"));
var ethereumjs_wallet_1 = __importDefault(require("ethereumjs-wallet"));
var test_provider_1 = require("@rarible/test-provider");
// @ts-ignore
var rpc_1 = __importDefault(require("web3-provider-engine/subproviders/rpc"));
var types_1 = require("@rarible/types");
function createE2eProvider(pk) {
    if (pk === void 0) { pk = types_1.randomWord(); }
    global.FormData = form_data_1.default;
    var provider = new web3_provider_engine_1.default();
    var wallet = new ethereumjs_wallet_1.default(Buffer.from(fixPK(pk), "hex"));
    provider.addProvider(new test_provider_1.TestSubprovider(wallet, { networkId: 17, chainId: 17 }));
    provider.addProvider(new rpc_1.default({ rpcUrl: "https://node-e2e.rarible.com" }));
    beforeAll(function () {
        provider.start();
    });
    afterAll(function () {
        provider.stop();
    });
    return {
        provider: provider,
        wallet: wallet,
    };
}
exports.createE2eProvider = createE2eProvider;
function fixPK(pk) {
    if (pk.startsWith("0x")) {
        return pk.substring(2);
    }
    else {
        return pk;
    }
}
