// @rarible/protocol-ethereum-sdk/build/order/test/order.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TEST_ORDER_TEMPLATE = void 0;
var address_1 = require("@rarible/types/build/address");
var big_number_1 = require("@rarible/types/build/big-number");
var types_1 = require("@rarible/types");
exports.TEST_ORDER_TEMPLATE = {
    make: {
        assetType: {
            assetClass: "ERC721",
            contract: address_1.toAddress("0x0000000000000000000000000000000000000001"),
            tokenId: big_number_1.toBigNumber("10"),
        },
        value: big_number_1.toBigNumber("10"),
    },
    maker: address_1.toAddress("0x0000000000000000000000000000000000000002"),
    take: {
        assetType: {
            assetClass: "ERC721",
            contract: address_1.toAddress("0x0000000000000000000000000000000000000001"),
            tokenId: big_number_1.toBigNumber("10"),
        },
        value: big_number_1.toBigNumber("10"),
    },
    salt: types_1.toWord("0x000000000000000000000000000000000000000000000000000000000000000a"),
};
